api/client/build.go

```diff
@@ -131,10 +130,19 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		return fmt.Errorf("cannot canonicalize dockerfile path %s: %v", relDockerfile, err)
 	}
 
-	var includes = []string{"."}
+	var (
+		includes = []string{"."}
+		excludes []string
+	)
 
-	excludes, err := utils.ReadDockerIgnore(path.Join(contextDir, ".dockerignore"))
-	if err != nil {
+	f, err := os.Open(filepath.Join(contextDir, ".dockerignore"))
```

Comment:
Because on the server side, I only have a reader when reading from a context: https://github.com/docker/docker/pull/16147/files#diff-ca102c238cfd962b0905c8009ad2c39bR42